### DISCUSSION TITLE
Strange training behavior related to the number of training steps

### DISCUSSION THREAD

**Discussion by mcourtiel** [deepmodeling/deepmd-kit#2442] [Q&A]
Hi,

I noticed a really strange behavior during the training of a model.
I trained it on the water monomer, with 160 000 structures of the water monomer extracted from a dynamic at 10K. I use 80% for the training and 20% for the validation and then testing. 
I trained two model from this data, but I use exactly the same sets of training and validation and exactly the same input.json, the only parameter I change between this two models is the “numb_steps” in the training part (300 000 for the first one and 600 000 for the second one).

When I train the model on 300 000 steps it looks really good with a RMSE of 3e-05 eV for the energies and 3e-03 eV/A for the forces and when I test the model on the validation and training data I have good results. 
But when I train this exactly same model on 600 000 steps something goes wrong, the force does not improve throughout the process (7e-02eV/A) and the RMSE for the energy (1e-03) is worse than the one of the previous model (train only on 300 000 steps). Another really strange behavior is that when I test it, either on a validation or a training set, I obtain the same result for the energies of every structure. And the forces are totally wrong.

Does anyone have any idea why this strange behavior happen ?

![test_model300000_energy](https://user-images.githubusercontent.com/122008786/230113735-06fb397a-2d3f-456b-a69e-efe1463c6a06.png)
![test_model600000_energy](https://user-images.githubusercontent.com/122008786/230113738-8eeb8ccd-6306-4fdc-8e23-1c04cc9eda6e.png)
![train_model300000_RMSE](https://user-images.githubusercontent.com/122008786/230113742-bb3aa286-504b-4df5-815f-fd26c5cf2885.png)
![train_model600000_RMSE](https://user-images.githubusercontent.com/122008786/230113746-b4877a72-bacd-44a3-b3ed-d1caa65290f5.png)
[300000_input.pdf](https://github.com/deepmodeling/deepmd-kit/files/11160030/300000_input.pdf)
[600000_input.pdf](https://github.com/deepmodeling/deepmd-kit/files/11160031/600000_input.pdf)

**Comment by wanghan-iapcm** [comment]
Please report the version of deepmd-kit and the hardware you used to train the models. Thank you .

**Comment by mcourtiel** [comment]
The version I use of deepmd-kit is v2.1.5.

For the hardware of my computer :

Packages:
    0: Intel Xeon E31220
Microarchitectures:
    4x Sandy Bridge
Cores:
    0: 1 processor (0), Intel Sandy Bridge
    1: 1 processor (1), Intel Sandy Bridge
    2: 1 processor (2), Intel Sandy Bridge
    3: 1 processor (3), Intel Sandy Bridge

I don't know if this infos are enought so you can have a look in the following file that contain all the infos of the training :
[Training_hardware_info.pdf](https://github.com/deepmodeling/deepmd-kit/files/11177466/Training_hardware_info.pdf)

Thank you

**Comment by wanghan-iapcm** [comment]
It seems that your `start_lr` is too large, please reduce it to e.g. 1e-3. 

When using a large lr, the training may become unstable, i.e. the error does not decrease. At the border of  stable-unstable lr, uncertain behavior may be observed among tasks, some tasks success by chance and others fail.

**Comment by mcourtiel** [comment]
Thanks a lot for your advice, it is working way better now.

The training is way better now.
![Mono_1000_ma_all](https://user-images.githubusercontent.com/122008786/231444375-5fdd995f-2175-4eb5-b994-37916e2531f8.PNG)
![Mono_1000_test009ener](https://user-images.githubusercontent.com/122008786/231444379-da8d0bd4-c391-4733-8991-dc67c53e673f.PNG)
